src/core/modules/Star.hpp

```diff
@@ -96,13 +96,18 @@ struct Star1 { // 28 byte
 	//
 	// qint32 dx0,dx1,plx   32
 private:
-	// Use an union so we can access the data as different types without
-	// aliasing issues.
-	union {
-		quint8  uint8[28];
-		quint16 uint16[14];
-		qint32  int32[7];
-	} d;
+	struct Data {
+		quint8  hip[3];	      // 3 bytes
+		quint8  componentIds; // 1 byte
+		qint32  x0;           // 4 bytes
+		qint32  x1;           // 4 bytes
+		quint8  b_v; 		  // 1 byte
+		quint8  vmag;         // 1 bytes
+		quint16 spInt;        // 2 bytes
+		qint32  dx0;          // 4 bytes
+		qint32  dx1;          // 4 bytes
+		qint32  plx;          // 4 bytes
+	} d;  // total is 28 bytes
```

Comment:
`static_assert` should be used to make really sure the size is 28 bytes.

Reply:
`static_assert` is already in the code

Reply:
Oh, indeed, didn't notice it.